src/consensus/pbft/pbft.h

```diff
@@ -104,16 +104,10 @@ namespace pbft
     std::unique_ptr<PbftEnclaveNetwork> pbft_network;
     std::unique_ptr<AbstractPbftConfig> pbft_config;
     std::unique_ptr<ClientProxy<kv::TxHistory::RequestID, void>> client_proxy;
-    kv::TxHistory::RequestCallbackHandler on_request;
     enclave::RPCSessions& rpcsessions;
     std::unique_ptr<raft::LedgerEnclave> ledger;
-
-    struct NodeConfiguration
-    {
-      NodeId node_id;
-      std::string host_name;
-      std::string port;
-    };
+    bool isleader;
```

Comment:
if we have local_id and my_info, we should have is_leader?

Reply:
I agree but then we can't have a method named is_leader()

Reply:
We probably should not have that, and have a role() method instead, returning a Role enum with PRIMARY, BACKUP, CANDIDATE, UNKNOWN.

Reply:
In which case will it be CANDIDATE or UNKNOWN in raft or in pbft? And when will it transition to PRIMARY or BACKUP?

Reply:
See: https://github.com/microsoft/CCF/blob/6efad0b64cc4dd3d0b20fd93228472ad7e8b329d/src/raft/raft.h#L54 and its usage. Unknown may or may not be needed at startup.

Reply:
Added as something to look into in next steps: #300